File: backend/routes/lms.py
"""
LMS API — /api/v1/lms
"""
import asyncio
import json
from typing import Optional
import anthropic as _anthropic
from fastapi import APIRouter, BackgroundTasks, Body, Depends, File, HTTPException, Query, UploadFile
from dependencies import get_current_user, require_manager_or_above, get_db
from config import settings
from models.lms import (
    CreateCourseRequest,
    UpdateCourseRequest,
    GenerateCourseRequest,
    EnrollRequest,
    UpdateProgressRequest,
    SubmitQuizRequest,
)
from services.lms_service import LmsService
from utils.ai_helpers import _strip_code_fence
from services.blob_storage import upload_blob, get_public_url

router = APIRouter()

# ── Published Courses (learner) ───────────────────────────────────────────────

@router.get("/courses")
async def list_courses(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    current_user: dict = Depends(get_current_user),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    user_id = current_user["sub"]
    return await LmsService.list_published_courses(conn, org_id, user_id, page, page_size)

# ── Course Management (manager+) ──────────────────────────────────────────────

@router.get("/courses/manage")
async def list_managed_courses(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    search: Optional[str] = Query(None),
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.list_managed_courses(conn, org_id, page, page_size, search)

@router.post("/courses")
async def create_course(
    body: CreateCourseRequest,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.create_course(conn, body, org_id, current_user["sub"])

@router.put("/courses/{course_id}/structure")
async def save_course_structure(
    course_id: str,
    body: dict = Body(...),
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    modules = body.get("modules", [])
    return await LmsService.save_course_structure(conn, course_id, org_id, modules)

@router.get("/courses/{course_id}")
async def get_course(
    course_id: str,
    current_user: dict = Depends(get_current_user),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.get_course(conn, course_id, org_id)

@router.post("/courses/{course_id}/regenerate-content")
async def regenerate_course_content(
    course_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    """Delete empty slides/quiz questions and re-generate them with Claude Haiku."""
    from services.db import row as _row, rows as _rows, execute as _execute, execute_returning as _execute_returning
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")

    # Fetch course (org ownership check)
    course = _row(
        conn,
        "SELECT id, title, description FROM courses WHERE id = %s AND organisation_id = %s",
        (course_id, org_id),
    )
    if not course:
        raise HTTPException(status_code=404, detail="Course not found.")
    course_title = course["title"]
    industry = course.get("description", "retail")

    # Fetch modules
    module_records = _rows(
        conn,
        "SELECT id, title, type FROM course_modules WHERE course_id = %s ORDER BY display_order",
        (course_id,),
    )
    if not module_records:
        raise HTTPException(status_code=400, detail="Course has no modules to regenerate.")

    # Step 1: Generate content with Claude Haiku FIRST — if this fails, nothing is deleted
    if not settings.anthropic_api_key:
        raise HTTPException(status_code=503, detail="AI content generation is not configured.")

    client = _anthropic.Anthropic(api_key=settings.anthropic_api_key)
    module_outline = [{"title": m["title"], "type": m["type"]} for m in module_records]

    def _call():
        return client.messages.create(
            model="claude-haiku-4-5-20251001",
            max_tokens=4000,
            messages=[{"role": "user", "content": (
                f"Generate training content for a course titled '{course_title}'. "
                f"Modules: {json.dumps(module_outline)}. "
                f"For each 'slides' module: write 3-4 slides. Each slide MUST have exactly two keys: \"title\" (short heading) and \"body\" (2-3 sentence explanation of a key concept). Do NOT use any other key name — only \"title\" and \"body\". "
                f"For each 'quiz' or 'video' module: write 3-4 multiple_choice quiz questions with 4 options (exactly 1 correct), and a brief explanation. "
                f"Respond ONLY with JSON array matching the module order: "
                f'[{{"type": "slides", "slides": [{{"title": "...", "body": "..."}}]}}, '
                f'{{"type": "quiz", "questions": [{{"question": "...", "question_type": "multiple_choice", "options": [{{"text": "...", "is_correct": true}}], "explanation": "..."}}]}}]'
            )}],
        )

    try:
        resp = await asyncio.to_thread(_call)
        content_text = "".join(b.text for b in resp.content if hasattr(b, "text"))
        content_list = json.loads(_strip_code_fence(content_text))
    except Exception as exc:
        raise HTTPException(status_code=500, detail=f"AI content generation failed: {exc}")

    # Step 2: Only now that generation succeeded, delete existing slides/quiz questions
    for mod in module_records:
        if mod["type"] == "slides":
            _execute(conn, "DELETE FROM course_slides WHERE module_id = %s", (mod["id"],))
        else:
            _execute(conn, "DELETE FROM quiz_questions WHERE module_id = %s", (mod["id"],))

    # Step 3: Insert new content
    for idx, mod in enumerate(module_records):
        if idx >= len(content_list):
            break
        mod_content = content_list[idx]
        if mod["type"] == "slides":
            for s_order, slide in enumerate(mod_content.get("slides") or []):
                _execute(
                    conn,
                    "INSERT INTO course_slides (module_id, title, body, display_order) VALUES (%s, %s, %s, %s)",
                    (
                        mod["id"],
                        slide.get("title", f"Slide {s_order + 1}"),
                        slide.get("body") or slide.get("content") or slide.get("text") or slide.get("description") or "",
                        s_order,
                    ),
                )
        else:
            for q_order, q in enumerate(mod_content.get("questions") or []):
                qtype = q.get("question_type", "multiple_choice")
                if qtype not in ("multiple_choice", "true_false", "image_based"):
                    qtype = "multiple_choice"
                _execute(
                    conn,
                    "INSERT INTO quiz_questions (module_id, question, question_type, options, explanation, display_order) VALUES (%s, %s, %s, %s::jsonb, %s, %s)",
                    (
                        mod["id"],
                        q.get("question", "Question"),
                        qtype,
                        json.dumps(q.get("options", [])),
                        q.get("explanation", ""),
                        q_order,
                    ),
                )

    return {"ok": True, "message": "Content regenerated successfully."}


@router.put("/courses/{course_id}")
async def update_course(
    course_id: str,
    body: UpdateCourseRequest,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.update_course(conn, course_id, body, org_id)

@router.delete("/courses/{course_id}")
async def delete_course(
    course_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.delete_course(conn, course_id, org_id)

@router.post("/courses/{course_id}/publish")
async def publish_course(
    course_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.publish_course(conn, course_id, org_id)

@router.get("/locations")
async def list_org_locations(
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.list_org_locations(conn, org_id)

@router.get("/courses/{course_id}/enrollable-users")
async def list_enrollable_users(
    course_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.list_enrollable_users(conn, course_id, org_id)

@router.get("/courses/{course_id}/enrollment-stats")
async def get_enrollment_stats(
    course_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.get_enrollment_stats(conn, course_id, org_id)

@router.post("/courses/{course_id}/duplicate")
async def duplicate_course(
    course_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.duplicate_course(conn, course_id, org_id, current_user["sub"])

@router.post("/courses/{course_id}/unpublish")
async def unpublish_course(
    course_id: str,
    body: dict = Body(default={}),
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    cancel = body.get("cancel_enrollments", False)
    return await LmsService.unpublish_course(conn, course_id, org_id, cancel)

# ── AI Generation ─────────────────────────────────────────────────────────────

@router.post("/courses/generate")
async def generate_course(
    body: GenerateCourseRequest,
    background_tasks: BackgroundTasks,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    created_by = current_user["sub"]
    job = await LmsService.start_ai_generation(conn, body, org_id, created_by)
    background_tasks.add_task(
        LmsService.process_generation_job,
        job["id"], body, org_id, created_by,
    )
    return job

@router.get("/courses/generate/{job_id}")
async def get_generation_job(
    job_id: str,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.get_ai_job(conn, job_id, org_id)

# ── Enrollments ───────────────────────────────────────────────────────────────

@router.get("/enrollments/my")
async def my_enrollments(
    current_user: dict = Depends(get_current_user),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.my_enrollments(conn, current_user["sub"], org_id)

@router.get("/enrollments")
async def list_enrollments(
    course_id: Optional[str] = Query(None),
    user_id: Optional[str] = Query(None),
    status: Optional[str] = Query(None),
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.list_enrollments(conn, org_id, course_id, user_id, status, page, page_size)

@router.post("/enrollments")
async def enroll_users(
    body: EnrollRequest,
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.enroll_users(conn, body, org_id, current_user["sub"])

@router.get("/enrollments/{enrollment_id}")
async def get_enrollment(
    enrollment_id: str,
    current_user: dict = Depends(get_current_user),
    conn=Depends(get_db),
):
    return await LmsService.get_enrollment_with_progress(conn, enrollment_id, current_user["sub"])

@router.post("/enrollments/{enrollment_id}/progress")
async def update_progress(
    enrollment_id: str,
    body: UpdateProgressRequest,
    current_user: dict = Depends(get_current_user),
    conn=Depends(get_db),
):
    return await LmsService.update_progress(conn, enrollment_id, body, current_user["sub"])

@router.post("/enrollments/{enrollment_id}/quiz/submit")
async def submit_quiz(
    enrollment_id: str,
    body: SubmitQuizRequest,
    current_user: dict = Depends(get_current_user),
    conn=Depends(get_db),
):
    return await LmsService.submit_quiz(conn, enrollment_id, body, current_user["sub"])

# ── Analytics ─────────────────────────────────────────────────────────────────

@router.get("/analytics/completion")
async def analytics_completion(
    current_user: dict = Depends(require_manager_or_above),
    conn=Depends(get_db),
):
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    return await LmsService.get_analytics_completion(conn, org_id)

# ── File Upload ────────────────────────────────────────────────────────────────

@router.post("/upload")
async def upload_training_file(
    file: UploadFile = File(...),
    current_user: dict = Depends(require_manager_or_above),
):
    import uuid, time
    org_id = (current_user.get("app_metadata") or {}).get("organisation_id")
    user_id = current_user["sub"]

    # Restrict to safe media types — reject anything that is not a video,
    # image, or PDF to prevent arbitrary file uploads.
    allowed_prefixes = ("video/", "image/", "application/pdf")
    content_type = file.content_type or ""
    if not any(content_type.startswith(prefix) for prefix in allowed_prefixes):
        raise HTTPException(status_code=422, detail="Unsupported file type")

    content = await file.read()
    if len(content) > 100 * 1024 * 1024:
        raise HTTPException(status_code=413, detail="File too large (max 100MB)")

    ext = (file.filename or "file").rsplit(".", 1)[-1].lower()
    suffix = uuid.uuid4().hex[:8]
    timestamp = int(time.time())
    storage_path = f"{org_id}/{user_id}/{timestamp}-{suffix}.{ext}"

    upload_blob("training-media", storage_path, content, file.content_type or "application/octet-stream")
    public_url = get_public_url("training-media", storage_path)
    return {"url": public_url}

File: frontend/services/lms.ts
import { apiFetch } from "./api/client";

export interface Course {
  id: string;
  organisation_id: string;
  created_by: string;
  title: string;
  description: string | null;
  thumbnail_url: string | null;
  estimated_duration_mins: number | null;
  passing_score: number;
  max_retakes: number | null;
  cert_validity_days: number | null;
  is_mandatory: boolean;
  target_roles: string[];
  target_location_ids: string[];
  is_published: boolean;
  was_published: boolean;
  is_active: boolean;
  ai_generated: boolean;
  language: string;
  version: number;
  parent_course_id: string | null;
  created_at: string;
  updated_at: string;
  course_modules?: CourseModule[];
  enrollment?: CourseEnrollment | null;
}

export interface CourseModule {
  id: string;
  course_id: string;
  title: string;
  module_type: "slides" | "video" | "pdf" | "quiz";
  content_url: string | null;
  display_order: number;
  is_required: boolean;
  estimated_duration_mins: number | null;
  course_slides?: CourseSlide[];
  quiz_questions?: QuizQuestion[];
}

export interface CourseSlide {
  id: string;
  module_id: string;
  title: string | null;
  body: string | null;
  image_url: string | null;
  display_order: number;
}

export interface QuizOption {
  id: string;
  text: string;
  is_correct: boolean;
}

export interface QuizQuestion {
  id: string;
  module_id: string;
  question: string;
  question_type: "multiple_choice" | "true_false" | "image_based";
  image_url: string | null;
  options: QuizOption[];
  explanation: string | null;
  display_order: number;
}

export interface CourseEnrollment {
  id: string;
  course_id: string;
  user_id: string;
  status: "not_started" | "in_progress" | "passed" | "failed";
  score: number | null;
  attempt_count: number;
  started_at: string | null;
  completed_at: string | null;
  cert_issued_at: string | null;
  cert_expires_at: string | null;
  cert_url: string | null;
  current_module_id: string | null;
  courses?: Course;
}

export interface CreateCoursePayload {
  title: string;
  description?: string;
  passing_score?: number;
  max_retakes?: number;
  cert_validity_days?: number;
  is_mandatory?: boolean;
  target_roles?: string[];
  target_location_ids?: string[];
  language?: string;
  modules?: ModulePayload[];
}

export interface ModulePayload {
  title: string;
  module_type: "slides" | "video" | "pdf" | "quiz";
  display_order?: number;
  slides?: { title?: string; body?: string; image_url?: string; display_order?: number }[];
  questions?: { question: string; question_type: string; options: QuizOption[]; explanation?: string; display_order?: number }[];
}

export interface GenerateCoursePayload {
  input_type: "topic" | "url" | "document" | "video";
  input_data?: string;
  input_file_url?: string;
  target_role?: string;
  language?: string;
}

export interface LmsAnalytics {
  total_enrollments: number;
  passed: number;
  in_progress: number;
  not_started: number;
  failed: number;
  completion_rate: number;
}

// ── Courses ─────────────────────────────────────────────────────────────────

export async function listPublishedCourses(params?: { page?: number; page_size?: number }) {
  const q = new URLSearchParams();
  if (params?.page) q.set("page", String(params.page));
  if (params?.page_size) q.set("page_size", String(params.page_size));
  return apiFetch<{ items: Course[]; total_count: number; page: number; page_size: number }>(
    `/api/v1/lms/courses${q.toString() ? `?${q}` : ""}`
  );
}

export async function listManagedCourses(params?: { page?: number; page_size?: number; search?: string }) {
  const q = new URLSearchParams();
  if (params?.page) q.set("page", String(params.page));
  if (params?.page_size) q.set("page_size", String(params.page_size));
  if (params?.search) q.set("search", params.search);
  return apiFetch<{ items: Course[]; total_count: number; page: number; page_size: number }>(
    `/api/v1/lms/courses/manage${q.toString() ? `?${q}` : ""}`
  );
}

export async function getCourse(courseId: string) {
  return apiFetch<Course>(`/api/v1/lms/courses/${courseId}`);
}

export async function createCourse(body: CreateCoursePayload) {
  return apiFetch<Course>(`/api/v1/lms/courses`, { method: "POST", body: JSON.stringify(body) });
}

export async function updateCourse(courseId: string, body: Partial<CreateCoursePayload>) {
  return apiFetch<Course>(`/api/v1/lms/courses/${courseId}`, { method: "PUT", body: JSON.stringify(body) });
}

export async function publishCourse(courseId: string) {
  return apiFetch<{ success: boolean }>(`/api/v1/lms/courses/${courseId}/publish`, { method: "POST" });
}

export interface EnrollableUser {
  id: string;
  full_name: string;
  role: string;
  location_id: string | null;
  enrollment_status: "not_started" | "in_progress" | "passed" | "failed" | null;
}

export interface OrgLocation {
  id: string;
  name: string;
}

export async function listEnrollableUsers(courseId: string) {
  return apiFetch<EnrollableUser[]>(`/api/v1/lms/courses/${courseId}/enrollable-users`);
}

export async function listOrgLocations() {
  return apiFetch<OrgLocation[]>(`/api/v1/lms/locations`);
}

export async function getEnrollmentStats(courseId: string) {
  return apiFetch<{ active_count: number; completed_count: number }>(
    `/api/v1/lms/courses/${courseId}/enrollment-stats`
  );
}

export async function unpublishCourse(courseId: string, cancelEnrollments: boolean) {
  return apiFetch<{ success: boolean }>(`/api/v1/lms/courses/${courseId}/unpublish`, {
    method: "POST",
    body: JSON.stringify({ cancel_enrollments: cancelEnrollments }),
  });
}

export async function deleteCourse(courseId: string) {
  return apiFetch<{ success: boolean }>(`/api/v1/lms/courses/${courseId}`, { method: "DELETE" });
}

export async function duplicateCourse(courseId: string) {
  return apiFetch<{ id: string }>(`/api/v1/lms/courses/${courseId}/duplicate`, { method: "POST" });
}

export async function generateCourse(body: GenerateCoursePayload) {
  return apiFetch<{ id: string; status: string }>(`/api/v1/lms/courses/generate`, { method: "POST", body: JSON.stringify(body) });
}

export async function uploadTrainingFile(file: File): Promise<{ url: string }> {
  const formData = new FormData();
  formData.append("file", file);
  return apiFetch<{ url: string }>("/api/v1/lms/upload", {
    method: "POST",
    body: formData,
    rawBody: true,
  });
}

export async function getGenerationJob(jobId: string) {
  return apiFetch<{ id: string; status: string; result_course_id: string | null; error_message: string | null }>(
    `/api/v1/lms/courses/generate/${jobId}`
  );
}

// ── Enrollments ─────────────────────────────────────────────────────────────

export async function getMyEnrollments() {
  return apiFetch<CourseEnrollment[]>(`/api/v1/lms/enrollments/my`);
}

export async function listEnrollments(params?: { course_id?: string; user_id?: string; status?: string; page?: number; page_size?: number }) {
  const q = new URLSearchParams();
  if (params?.course_id) q.set("course_id", params.course_id);
  if (params?.user_id) q.set("user_id", params.user_id);
  if (params?.status) q.set("status", params.status);
  if (params?.page) q.set("page", String(params.page));
  if (params?.page_size) q.set("page_size", String(params.page_size));
  return apiFetch<{ items: CourseEnrollment[]; total_count: number }>(
    `/api/v1/lms/enrollments${q.toString() ? `?${q}` : ""}`
  );
}

export async function enrollUsers(courseId: string, userIds: string[], isMandatory = false) {
  return apiFetch<{ enrolled: number; skipped: number }>(`/api/v1/lms/enrollments`, {
    method: "POST",
    body: JSON.stringify({ course_id: courseId, user_ids: userIds, is_mandatory: isMandatory }),
  });
}

export async function updateProgress(enrollmentId: string, moduleId: string, status: string, timeSpent?: number) {
  return apiFetch<{ success: boolean }>(`/api/v1/lms/enrollments/${enrollmentId}/progress`, {
    method: "POST",
    body: JSON.stringify({ module_id: moduleId, status, time_spent_seconds: timeSpent }),
  });
}

export async function submitQuiz(enrollmentId: string, moduleId: string, answers: { question_id: string; selected_option: string }[]) {
  return apiFetch<{ score: number; passed: boolean; correct: number; total: number; attempt_number: number }>(
    `/api/v1/lms/enrollments/${enrollmentId}/quiz/submit`,
    { method: "POST", body: JSON.stringify({ module_id: moduleId, answers }) }
  );
}

// ── Analytics ────────────────────────────────────────────────────────────────

export async function getLmsAnalytics() {
  return apiFetch<LmsAnalytics>(`/api/v1/lms/analytics/completion`);
}

export async function saveCourseStructure(courseId: string, modules: ModulePayload[]) {
  return apiFetch<Course>(`/api/v1/lms/courses/${courseId}/structure`, {
    method: "PUT",
    body: JSON.stringify({ modules }),
  });
}

export interface ModuleProgress {
  id: string;
  enrollment_id: string;
  module_id: string;
  status: "not_started" | "in_progress" | "completed";
  started_at: string | null;
  completed_at: string | null;
  time_spent_seconds: number;
}

export interface EnrollmentWithProgress {
  enrollment: CourseEnrollment;
  course: Course;
  module_progress: ModuleProgress[];
}

export async function getEnrollmentWithProgress(enrollmentId: string) {
  return apiFetch<EnrollmentWithProgress>(`/api/v1/lms/enrollments/${enrollmentId}`);
}

// ── AI endpoints ──────────────────────────────────────────────────────────────

export function generateQuiz(body: {
  course_id: string;
  slides_content: string[];
  num_questions?: number;
}): Promise<{ questions: Array<{ question: string; options: string[]; correct_index: number; explanation: string }> }> {
  return apiFetch("/api/v1/ai/generate-quiz", {
    method: "POST",
    body: JSON.stringify(body),
  });
}

export function translateCourse(body: {
  course_id: string;
  target_language: string;
  content: object;
}): Promise<object> {
  return apiFetch("/api/v1/ai/translate-course", {
    method: "POST",
    body: JSON.stringify(body),
  });
}

export function getKnowledgeGaps(body: {
  wrong_answers: Array<{ question: string; chosen: string; correct: string; course_title: string }>;
}): Promise<{ gaps: Array<{ topic: string; description: string; severity: "low" | "medium" | "high"; recommended_action: string }> }> {
  return apiFetch("/api/v1/ai/knowledge-gaps", {
    method: "POST",
    body: JSON.stringify(body),
  });
}

export function getLearningPath(body: {
  role: string;
  completed_courses: string[];
  quiz_scores: Record<string, number>;
  available_courses: Array<{ id: string; title: string; type: string }>;
}): Promise<{ recommended: Array<{ course_id: string; reason: string; priority: number }> }> {
  return apiFetch("/api/v1/ai/learning-path", {
    method: "POST",
    body: JSON.stringify(body),
  });
}
